### PR TITLE
Update spatial autocor and other things

### PR DESCRIPTION
**Pull request by lfranzen**
Working towards update 1.1.7 (still a few more things to do though).

Added:  

- Compatibility with parquet file format, used in Visium HD(?), when loading coordinate files using the `LoadSpatialCoordinates` function (used in `ReadVisiumData`).
- Optional alternative to compute p-values when calculating spatial autocorrelation scores using `CorSpatialFeatures`. The p-values are computed using the Student t Distribution, and adjusted p-values are calculated using the Benjamini & Hochberg (BH, or FDR) method.

Bug fixes:

- Dealt with an issue in `MapFeatures` and `MapLabels` if the specified column contains NA values and when using SeuratObject v.5.
- When running `RegionNeighbors` and specifying a column with labels that contains blankspaces, each blank space will be replaced with punctuation marks in the newly formed column names. A warning message will be displayed in these cases.

Other: 
Some other minor things, like fixing typos and updating tests.

### CODE REVIEW

File: R/load_data.R [hunk 1]
```diff
@@ -245,8 +245,9 @@ LoadSpatialCoordinates <- function (
       if (!requireNamespace("arrow", quietly = TRUE)) {
         abort(glue("Package {cli::col_br_magenta('arrow')} is required. Please install it with: \n",
                    "install.packages('arrow')"))
+      } else if (requireNamespace("arrow", quietly = TRUE)) {
+        coords <- arrow::read_parquet(file = coordinatefiles[i], as_data_frame = TRUE, mmap = TRUE)
       }
```
Comment (ludvigla) [inline R/load_data.R:250]: I would move the `requireNamespace("arrow", quietly = TRUE)` out of the if statement. If the package is missing the function will abort. And if it's installed, using arrow:: will attach the function so no need to run requireNamespace. 
So I would switch this:
````
else if (requireNamespace("arrow", quietly = TRUE)) {
        coords <- arrow::read_parquet(file = coordinatefiles[i], as_data_frame = TRUE, mmap = TRUE)
      }
````
to simply use:

````
coords <- arrow::read_parquet(file = coordinatefiles[i], as_data_frame = TRUE, mmap = TRUE)
````

File: R/neighborhood_analysis.R [hunk 2]
```diff
@@ -274,6 +274,10 @@ RegionNeighbors.Seurat <- function (
 
   # Add data to Seurat object
   nbs_rearranged <- do.call(bind_cols, lapply(column_labels, function(lbl) {
+    if (grepl("\\s", lbl)) {
+      new_lbl <- gsub(" ", "\\.", lbl)
+      cli_alert_warning("Blankspaces detected among column labels. New column name set to '{new_lbl}'.")
+    }
```
Comment (ludvigla) [inline R/neighborhood_analysis.R:280]: Perhaps it's better to be strict and abort is there are blankspaces detected. I think it's bad practice to use those but you should decide.

File: R/neighborhood_analysis.R [hunk 3]
```diff
@@ -282,10 +286,10 @@ RegionNeighbors.Seurat <- function (
       select(-barcode) |>
       setNames(nm = c("var1", "var2")) |>
       mutate_if(is.factor, as.character) |>
-      mutate(var2 = case_when((!is.na(var2)) & (var1 != var2) ~ paste0(column_key, lbl),
+      mutate(var2 = case_when((!is.na(var2)) & (var1 != var2) ~ paste0(column_key, gsub(" ", "\\.", lbl)),
                               TRUE ~ var2)) |>
       select(var2) |>
-      setNames(nm = paste0(column_key, lbl))
+      setNames(nm = paste0(column_key, gsub(" ", "\\.", lbl)))
```
Comment (ludvigla) [inline R/neighborhood_analysis.R:292]: Not entirely sure what this does, but aren't the new names already stored in `new_lbl`? Is there a reason to compute these again?

File: R/spatial_autocorrelation.R [hunk 4]
```diff
@@ -197,9 +201,19 @@ CorSpatialFeatures.default <- function (
...
+    }
+    
     # Summarize results
-    results <- tibble(gene = names(spatial_autocorrelation), cor = spatial_autocorrelation) |>
-      arrange(-cor)
+    results <- tibble(gene = names(spatial_autocorrelation), cor = spatial_autocorrelation)
+    if (calculate_pvalue) {
+      results <- results |> 
+        add_column(pval = pvals) |> 
+        add_column(adj.pval = p.adjust(pvals, method = "BH"))
```
Comment (ludvigla) [inline R/spatial_autocorrelation.R:214]: Perhaps there should be an option to set the MH adjustment method?

**Review by ludvigla — CHANGES_REQUESTED** [review]

File: R/spatial_autocorrelation.R (hunk 4 above)
Comment (lfranzen) [reply R/spatial_autocorrelation.R:214]: I thought about that as well, but it's such an easy thing to do afterwards yourself so I might leave it like this (for now at least).

File: R/load_data.R (hunk 1 above)
Comment (lfranzen) [reply R/load_data.R:250]: Alright, thanks that makes sense. Then I'll go back to how it was before. We're still getting a warning about this though when you run the checks on the package, so there's something about how this is being called that the checks doesn't like.

File: R/neighborhood_analysis.R (hunk 2 above)
Comment (lfranzen) [reply R/neighborhood_analysis.R:280]: Now I changed it to what you suggested instead to perform a check and abort if any blank spaces are found. I'll push it soon

File: R/neighborhood_analysis.R (hunk 3 above)
Comment (lfranzen) [reply R/neighborhood_analysis.R]: Rewrote this entirely to stop function if blank spaces are detected

File: R/load_data.R (hunk 1 above)
Comment (ludvigla) [reply R/load_data.R]: Sorry, I wasn't clear. This is what it should look like:
````
if (!requireNamespace("arrow", quietly = TRUE)) {
      abort(glue("Package {cli::col_br_magenta('arrow')} is required. Please install it with: \n",
                "install.packages('arrow')"))
} 
coords <- arrow::read_parquet(file = coordinatefiles[i], as_data_frame = TRUE, mmap = TRUE)
````
I meant to say that this part should be moved out: `coords <- arrow::read_parquet(file = coordinatefiles[i], as_data_frame = TRUE, mmap = TRUE)
`

**Pull request merged**